### PR TITLE
test: Add /dev/random entropy measurement

### LINKED ISSUE #1543: entropy/randomness: tests need improving
Enabling the kernel `CONFIG_RANDOM_TRUST_CPU` option over in
https://github.com/kata-containers/packaging/pull/482
has highlighted that when we added the [entropy test](https://github.com/kata-containers/tests/pull/887) to the CI, maybe it is not quite testing what we thought, and maybe we really now need to test the availability of random data, not just the amount of entropy available?

Maybe we can do something along the lines of:

```bash
$ dd if=/dev/random of=/dev/null bs=4b count=1000000
```

and time it - and ensure it finishes in 'a reasonable time' ?
I have also wondered if we can run something like the openssl benchmarks, maybe they consume randomness?

### PR DESCRIPTION
**Pull request by GabyCT**
We need to measure the availability of random data.

Fixes #1543

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

File: integration/entropy/entropy_time.bats [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+@test "entropy for /dev/random" {
+	output_file=$(mktemp)
+	block_size="4b"
+	entropy_time="20"
+	copy_number="100000"
+
+	{ TIMEFORMAT="%R"; time dd if=/dev/random of=/dev/null bs="$block_size" count="$copy_number" 2>/dev/null ; } 2> "$output_file"
+	measured_entropy=$(cat $output_file | cut -d '.' -f1)
+	[ "$measured_entropy" -le "$entropy_time" ]
+}
```
Comment (jcvenegas) [inline integration/entropy/entropy_time.bats]: Missing indentation

File: integration/entropy/entropy_time.bats [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+setup() {
+	clean_env
+	# Check that processes are not running
+	run check_processes
+	echo "$output"
+	[ "$status" -eq 0 ]
+}
+
+@test "entropy for /dev/random" {
```
Comment (chavafg) [inline integration/entropy/entropy_time.bats]: this is testing that time for this operation is reasonable, right? maybe change the name to reflect that?

File: integration/entropy/entropy_time.bats [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	output_file=$(mktemp)
+	block_size="4b"
+	entropy_time="10"
```
Comment (chavafg) [inline integration/entropy/entropy_time.bats]: should we add a comment of why this is a good value?
also, can the variable name be changed to something like `expected_time` ?

File: integration/entropy/entropy_time.bats [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	copy_number="100000"
+
+	{ TIMEFORMAT="%R"; time dd if=/dev/random of=/dev/null bs="$block_size" count="$copy_number" 2>/dev/null ; } 2> "$output_file"
+	measured_entropy=$(cat $output_file | cut -d '.' -f1)
```
Comment (chavafg) [inline integration/entropy/entropy_time.bats]: `measured_time=...`?

File: integration/entropy/entropy_time.bats [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+	[ "$measured_entropy" -le "$entropy_time" ]
+}
+
+teardown() {
+	clean_env
+	sudo rm "$output_file"
```
Comment (chavafg) [inline integration/entropy/entropy_time.bats]: sudo shouldn't be needed here

**Comment by GabyCT** [comment]
@chavafg changes applied

File: Makefile [hunk 6]
```diff
@@ -150,6 +150,7 @@ entropy:
 	bash -f .ci/install_bats.sh
 	cd integration/entropy && \
 	bats entropy_test.bats
+	bats entropy_time.bats
```
Comment (grahamwhaley) [inline Makefile]: A niggle... is this right? I'm not sure the `cd` carries over to the 2nd line does it?
We seem to have a mix through the `Makefile` where some `bats` tests use the `cd && bats` idiom, and others use the `bats fullpath` idiom. I prefer the second (the fullpath) one?
Maybe make both these entropy bats lines `bats <fullpath to batsfile>` style?

**Comment by GabyCT** [comment]
@chavafg changes applied

**Comment by GabyCT** [comment]
@grahamwhaley changes applied

**Comment by grahamwhaley** [comment]
Thanks @GabyCT - need @chavafg to ack his change request I think.

File: integration/entropy/entropy_time.bats [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+	[ "$status" -eq 0 ]
+}
+
+@test "measured time for /dev/random" {
+	output_file=$(mktemp)
+	block_size="4b"
+	expected_time="20"
+	copy_number="100000"
+	image="busybox"
+	cmd="time dd if=/dev/random of=/dev/null bs=$block_size count=$copy_number"
```
Comment (chavafg) [inline integration/entropy/entropy_time.bats]: do we really need `time` ? I see that later you are doing `.. | grep seconds`, which comes from the output of `dd`

Comment (grahamwhaley) [reply integration/entropy/entropy_time.bats]: ah, if `dd` gives the time, then sure we can use that. I my home run local example I was thinking of the seconds from the time. heh.

**Comment by GabyCT** [comment]
@chavafg thanks for the feedback , changes applied

**Comment by chavafg** [comment]
thanks @GabyCT

**Pull request merged**